src/main/java/org/xbill/DNS/Socks5Proxy.java

```diff
@@ -0,0 +1,235 @@
+package org.xbill.DNS;
+
+import lombok.Getter;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+import java.nio.channels.SocketChannel;
+import java.util.Objects;
+
+@Getter
+public class Socks5Proxy {
```

Comment:
I'm not sure if there's a need for timeout handling?

Reply:
You can set timeouts on the Dante config. It should reply with "TTL expired" then. Don't know, how to do this properly. There might be the need to track and handle this. Or you add a retry mechanism.